### PR TITLE
repo-updater: Exclude soft deleted services from listing

### PR DESCRIPTION
**Pull request by keegancsmith**
Including soft deleted services leads to us syncing deleted
external services. Additionally I went through each call site of
ListExternalServices and every one requires only active external
services. Migrations can run against soft-deleted external services as
well. However, they are just as good without them.

Test plan: go test

Part of #2025

### CODE REVIEW

**Comment by tsenart** [comment]
As spoken out-of-band, we should make this consistent with `ListRepos` which needs to return soft-deleted repos. A possible solution we discussed is to optimise for the common case, which is likely not the current behaviour of returning soft deleted repos. We can do that by adding a `Deleted` field to the `StoreListXXXArgs` structs which, when true, would include deleted repos in the result set. This would require a change in the documentation of that type.

File: cmd/repo-updater/repos/store.go [hunk 1]
```diff
@@ -36,15 +36,22 @@ type StoreListReposArgs struct {
...
 	Kinds []string
+	// If true, includes deleted repos in the result set.
+	Deleted bool
 }
 
 // StoreListExternalServicesArgs is a query arguments type used by
 // the ListExternalServices method of Store implementations.
+//
+// Each defined argument must map to a disjunct (i.e. AND) filter predicate.
+// When zero-valued, an argument is ommited from the predicate set.
```
Comment (tsenart) [inline cmd/repo-updater/repos/store.go]:
```suggestion
```

File: cmd/repo-updater/repos/store.go [hunk 2]
```diff
@@ -36,15 +36,22 @@ type StoreListReposArgs struct {
...
 type StoreListExternalServicesArgs struct {
 	// IDs of external services to list.
 	IDs []int64
 	// Kinds of external services to list.
```
Comment (tsenart) [inline cmd/repo-updater/repos/store.go]:
```suggestion
	// Kinds of external services to list. When zero-valued, this is omitted from the predicate set.
```

File: cmd/repo-updater/repos/store.go [hunk 3]
```diff
@@ -36,15 +36,22 @@ type StoreListReposArgs struct {
...
+	Deleted bool
 }
 
 // StoreListExternalServicesArgs is a query arguments type used by
 // the ListExternalServices method of Store implementations.
+//
+// Each defined argument must map to a disjunct (i.e. AND) filter predicate.
+// When zero-valued, an argument is ommited from the predicate set.
 type StoreListExternalServicesArgs struct {
 	// IDs of external services to list.
```
Comment (tsenart) [inline cmd/repo-updater/repos/store.go]:
```suggestion
	// IDs of external services to list. When zero-valued, this is omitted from the predicate set.
```

File: cmd/repo-updater/repos/store.go [hunk 4]
```diff
@@ -28,23 +28,28 @@ type Store interface {
...
+// Each defined argument must map to a disjunct (i.e. AND) filter predicate.
 type StoreListExternalServicesArgs struct {
-	// IDs of external services to list.
+	// IDs of external services to list. When zero-valued, this is omitted from the predicate set.
 	IDs []int64
-	// Kinds of external services to list.
+	// Kinds of external services to list. When zero-valued, this is omitted from the predicate set.
 	Kinds []string
+	// If true, includes deleted services in the result set.
+	Deleted bool
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/store.go]: Given we don't use it, I'd rather just exclude it from StoreListExternalServiceArgs.

Comment (tsenart) [reply cmd/repo-updater/repos/store.go]: Good call. I'll remove it.

**Pull request merged**